test/com/amazon/ion/LoadBinaryIonReaderSystemProcessingTest.java

```diff
@@ -0,0 +1,31 @@
+package com.amazon.ion;
+
+public class LoadBinaryIonReaderSystemProcessingTest
+    extends DatagramIteratorSystemProcessingTest
+{
+    private byte[] myBytes;
+
+    @Override
+    protected void prepare(String text)
+    {
+        myMissingSymbolTokensHaveText = false;
+        myBytes = encode(text);
+    }
+
+    @Override
+    protected IonDatagram load() throws Exception
+    {
+        IonLoader loader = loader();
+        IonReader reader = getStreamingMode().newIonReader(system().getCatalog(), myBytes);
+        IonDatagram datagram = loader.load(reader);
+        reader.close();
+        return datagram;
+    }
+
+    @Override
+    protected int expectedLocalNullSlotSymbolId() {
+        // The spec allows for implementations to treat these malformed symbols as "null slots", and all "null slots"
+        // in local symbol tables as equivalent to symbol zero.
+        return getStreamingMode() == StreamingMode.NEW_STREAMING_INCREMENTAL ? 0 : 10;
```

Comment:
Why is this? I did a little poking around and don't get it yet. Is this a test case convention? I see this in more places.

Reply:
The spec [says](https://amzn.github.io/ion-docs/docs/symbols.html#local-symbol-tables) "Any SIDs that refer to null slots in a local symbol table are equivalent to symbol zero." However, unlike the new implementation, the existing binary reader implementation does not actually assign null slots symbol ID 0. It's a larger effort to fix this in the old implementation, and is left out of scope for this PR (see #151 ).

10 is hardcoded here because the method is only called in places where the null slot is at symbol ID 10; there’s nothing special about 10 with respect to null slots.

Reply:
> there’s nothing special about 10 with respect to null slots.

That's more what I was getting at. I had a hard time finding the significance of 10, in particular.

These tests and their data have behavior and specification split up in ways that make it difficult as a reader to assemble. I'd like to see a different factoring that lowers the barrier to entry. 

Nothing to do here, necessarily, except that while raising 10 as a special value in these tests does exercise a behavior of the new reader that performs to spec I'd rather see that behavior demonstrated in a specific and isolated test so that I don't have to drag the knowledge of "10" across boundaries.